flink-core/src/main/java/org/apache/flink/configuration/HeartbeatManagerOptions.java

```diff
@@ -50,6 +50,8 @@ public class HeartbeatManagerOptions {
     private static final String HEARTBEAT_RPC_FAILURE_THRESHOLD_KEY =
             "heartbeat.rpc-failure-threshold";
 
+    public static final int FAILED_RPC_DETECTION_DISABLED = -1;
```

Comment:
hm, I'm not feeling comfortable with adding production code (which I consider the default value to be) to the options definitions. The reason for this change was that flink-core doesn't have access to flink-runtime, wasn't it? Alternatively, we could move the HeartbeatManagerOptions into the flink-runtime package. That would require us to add the package that contains the options to [ConfigurationOptionLocator](https://github.com/apache/flink/blob/6cce68dcdc1baf4be2a9e1549983d010644b5ee3/flink-docs/src/main/java/org/apache/flink/docs/util/ConfigurationOptionLocator.java).
I understand that this is a more invasive change. But it feels more natural to have the options where the actual implementation is, anyway. WDYT?

Reply:
Looking at my comment you linked: I see how this can be misunderstood now. Sorry for that one. 8) I meant "implementation" instead of "production" code. My initial concern was that I couldn't find any other occurrence where we put constants next to the `ConfigOption` declarations. It seems like we usually have them closer to the implementation. Anyway, you have a point with your arguments.